pkg/storegateway/gateway_blocks_http.go

```diff
@@ -156,9 +197,105 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 		ShowDeleted: showDeleted,
 		ShowSources: showSources,
 		ShowParents: showParents,
+		ActionType:  action,
+		InfoText:    infoText,
 	}, blocksPageTemplate, req)
 }
 
+func (s *StoreGateway) BlocksWriteHandler(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	tenantID := vars["tenant"]
+	if tenantID == "" {
+		util.WriteTextResponse(w, "Tenant ID can't be empty")
+		return
+	}
+
+	if err := req.ParseForm(); err != nil {
+		util.WriteTextResponse(w, fmt.Sprintf("Can't parse form: %s", err))
+		return
+	}
+	actionType := req.Form.Get("action_type")
+	if actionType == "" {
+		actionType = string(ActionTypeNone) // Set the default value to "none"
+	}
+
+	if !isValidActionType(ActionType(actionType)) {
+		util.WriteTextResponse(w, fmt.Sprintf("Invalid Action Type: %s\n", actionType))
+		return
+	}
+
+	// Perform the action on selected blocks
+	action := ActionType(actionType)
```

Comment:
Opinionated, but I find it weird that we convert a typed const to a string, and then we validate that it's the correct value.

How about we only validate when there's input?

```suggestion
	action := ActionTypeNone
	if actionType := req.Form.Get("action_type"); actionType != "" {
		if !isValidActionType(ActionType(actionType)) {
			util.WriteTextResponse(w, fmt.Sprintf("Invalid Action Type: %s\n", actionType))
			return
		}	
		action = ActionType(actionType)
	}
```

Secondary nit is: `isValidActionType` should accept a `string`, not an `ActionType`, this way we can put boundaries and assume that anything of type `ActionType` is valid.